### PR TITLE
chore(*): Improve @clerk/backend DX [Part 1 - Refactoring]

### PR DESCRIPTION
**Pull request by dimkl**
## Description

Apply some internal refactoring and restructuring to `@clerk/backend`.
Review it per commit for more information.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/backend/src/runtime.ts [hunk 1]
```diff
@@ -11,12 +11,18 @@
  *
  * TODO: Support TS runtime modules
  */
-
-// @ts-ignore - These are package subpaths
-import crypto from '#crypto';
+let webcrypto;
```
Comment (nikosdouvlis) [inline packages/backend/src/runtime.ts:14]: Why did you remove the subpath import here? Are you trying to make Jest work without configuring a `moduleMapper`? If so, Jest claims that that subpath imports work since this release: https://github.com/jestjs/jest/releases/tag/v29.4.0 :) #til 

I haven't tested the following, but this is what I had in mind for `#crypto`: 

`node/crypto.mjs`:
```
export { webcrypto } from 'node:crypto';
```

`node/crypto.js`:
```
module.exports.webcrypto = require('node:crypto').webcrypto;
```

`package.json`:
```
  "imports": {
    "#crypto": {
      "edge-light": "./dist/runtime/browser/crypto.mjs",
      "worker": "./dist/runtime/browser/crypto.mjs",
      "browser": "./dist/runtime/browser/crypto.mjs",
      "node": {
        "require": "./dist/runtime/node/crypto.js",
        "import": "./dist/runtime/node/crypto.mjs"
      },
      "default": "./dist/runtime/browser/crypto.mjs"
    }
  },
```

`runtime.ts`:
```
import { webcrypto } from '#crypto';
...
```

**Review by nikosdouvlis — APPROVED** [review]
This makes sense to me, one minor comment about the crypto imports

File: packages/backend/src/runtime.ts (hunk 1 above)
Comment (dimkl) [reply packages/backend/src/runtime.ts]: Change reverted and suggestion applied.
